packages/netsuite-adapter/src/adapter.ts

```diff
@@ -57,6 +58,47 @@ export interface NetsuiteAdapterParams {
   config: NetsuiteConfig
 }
 
+export const findDependingInstancesFromRefs = (instance: InstanceElement): InstanceElement[] => {
+  const visitedIdToInstance = new Map<string, InstanceElement>()
+  const isRefToServiceId = (topLevelParent: InstanceElement, elemId: ElemID): boolean => {
+    const fieldType = getFieldType(topLevelParent.type, [...elemId.createTopLevelParentID().path])
+    return isPrimitiveType(fieldType) && fieldType.isEqual(BuiltinTypes.SERVICE_ID)
+  }
+
+  const createDependingElementsCallback: TransformFunc = ({ value }) => {
+    if (isReferenceExpression(value)) {
+      const { topLevelParent, elemId } = value
+      if (isInstanceElement(topLevelParent)
+        && !visitedIdToInstance.has(topLevelParent.elemID.getFullName())
+        && isRefToServiceId(topLevelParent, elemId)
+        && elemId.adapter === NETSUITE) {
+        visitedIdToInstance.set(topLevelParent.elemID.getFullName(), topLevelParent)
+      }
+    }
+    return value
+  }
+
+  transformElement({
+    element: instance,
+    transformFunc: createDependingElementsCallback,
+    strict: true,
+  })
+  return wu(visitedIdToInstance.values()).toArray()
+}
+
+/**
+ * This method runs recursively on all references of the changedInstance to identify dependencies
+ * and then deploy the depending instances as well as part of the SDF project
+ */
+const getAllDependingInstances = (changedInstance: InstanceElement,
+  visitedDependingIds: Set<string>): InstanceElement[] => {
+  const dependingInstances = findDependingInstancesFromRefs(changedInstance)
+    .filter(instance => !visitedDependingIds.has(instance.elemID.getFullName()))
+  dependingInstances.forEach(instance => visitedDependingIds.add(instance.elemID.getFullName()))
+  return _.flatten(dependingInstances.map(instance =>
+    [instance, ...getAllDependingInstances(instance, visitedDependingIds)]))
```

Comment:
style nit: prefer built in features over lodash where possible
```suggestion
  return dependingInstances.flatMap(
    instance =>[instance, ...getAllDependingInstances(instance, visitedDependingIds)]
  )
```

Reply:
Also, maybe change the behavior of `getAllDependingInstances` to return the instance itself as well?
```suggestion
  return [changedInstance, ...dependingInstances.flatMap(
    instance => getAllDependingInstances(instance, visitedDependingIds)
  )]
```
that way at the top level we can just have 
```
const instancesToDeploy = changedInstances.flatMap(instance => getAllDependingInstances(instance, visitedDependingIds))
```